libs/core/src/lib/radio/radio-button/radio-button.component.ts

```diff
@@ -0,0 +1,217 @@
+     * The field is mandatory when working with reactive forms
+     */
+    @Input()
+    set selectedValue(v: any) {
+        this.actualValue = v;
+    }
+
+    private _name: string;
+    /** The name of the radio button
+     * The field is mandatory
+     */
+    @Input()
+    set name(v: string) {
+        this._name = v;
+    }
+
+    /** @hidden */
+    get name(): string {
+        return this._name;
+    }
+
+    get checked(): boolean {
+        if (this.value === undefined) {
+            return false;
+        }
+        return this.actualValue === this.value;
+    }
+
+    /**
+     * Set uniqueId to a radio button
+     */
+    @Input()
+    id: string = `radio-id-${uniqueId++}`;
+
+    /** Value field stores information about holding value by radio button
+     * The field is mandatory
+     */
+    @Input()
+    value: any;
+
+    // ControlValueAccessor implementation
+    /** @hidden */
+    onChange: any = () => { };
+
+    /** @hidden */
+    onTouched: any = () => { };
+
+    /** @hidden */
+    registerOnChange(fn: (selected: any) => { void }): void {
+        this.onChange = fn;
+    }
+
+    /** @hidden */
+    registerOnTouched(fn: any): void {
+        this.onTouched = fn;
+    }
+
+    /** @hidden */
+    setDisabledState(isDisabled: boolean): void {
+        this.disabled = isDisabled;
+        this.changeDetectionRef.detectChanges();
+    }
+
+    /** @hidden */
+    writeValue(value: any): void {
+        this.valueChange(value);
+    }
+    // End implementation
+
+    /** @hidden */
+    labelClicked(): void {
+        this.valueChange(this.value);
+        this._setFocusOnNativeElement();
+    }
+
+    /** @hidden */
+    valueChange(value: any): void {
+        this.actualValue = value;
+
+        this._setNativeElementCheckedState();
+
+        this.changeDetectionRef.detectChanges();
+        this.onChange(value);
+    }
+
+    /** @hidden */
+    constructor(private changeDetectionRef: ChangeDetectorRef) { }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this.buildComponentCssClass();
+        this._checkMandatoryFields();
+    }
+
+
+    /** This method is responsible for building a css class based on current state
+     *  It is implementation of CssClassBuilder interface and
+     *  should be used with @applyCssClass decorator
+     */
+    @applyCssClass
+    buildComponentCssClass(): string {
+        return [
+            'fd-radio',
+            this._compact ? 'fd-radio--compact' : '',
+            this._state !== 'default' ? `is-${this._state}` : ''
+        ].join(' ');
+    }
+
+    /** @hidden */
+    elementRef(): ElementRef<any> {
+        return this.inputElement;
+    }
+
+    /** @hidden */
+    private _checkMandatoryFields() {
+        if (!this.name) {
+            throw 'name field is required';
+        }
+        if (!this.value) {
+            throw 'value field is required';
+        }
+    }
+
+    /** @hidden */
+    private _setFocusOnNativeElement() {
```

Comment:
type missing

Reply:
done, I'll check if return type can be added to lint